### ISSUE TITLE
Decouple accumulated gradient from logging

### ISSUE THREAD

**Issue by Icemole** [rwth-i6/returnn#1820]
Hi, I've been profiling my training with `accum_grad_multiple_step = 4`, and I've found some `<listcomp>` operation taking a non-negligible part of the training time:

```
Total Samples 197500
GIL: 1.00%, Active: 500.00%, Threads: 8

  %Own   %Total  OwnTime  TotalTime  Function (filename)
  0.00%   0.00%   667.0s    667.0s   _engine_run_backward (torch/autograd/graph.py)
 94.00%  94.00%   508.8s    509.4s   raw_dict_to_extern_data (returnn/returnn/torch/data/extern_data.py)
  0.00%   0.00%   355.9s    363.1s   <listcomp> (returnn/returnn/torch/engine.py)
  3.00%   3.00%   279.2s    279.2s   all_reduce (torch/distributed/distributed_c10d.py)
  0.00%   0.00%   22.34s    22.54s   forward (torch/nn/modules/linear.py)
```

After some stack trace dumping, I found that the `<listcomp>` operation was `_get_total_grad_norm`, which is gradient accumulation for insertion into `losses_dict` and posterior logging. You can find the specific operation [here](https://github.com/rwth-i6/returnn/blob/08f30da0996ddc88b5695939e528c420e85dc9a3/returnn/torch/engine.py#L1767).

If this is only for insertion into the `losses_dict` and posterior logging, since logging takes a very long time, I propose decoupling logging from actual computation by adding a `log_grad_norm_epoch_interval` or similar, which could be increased if required so that the printing doesn't take that much time.

What do you think?
